src/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -337,6 +337,10 @@ public function isMethodStaticCallOrClassMethodObjectType(Node $node, ObjectType
             return true;
         }
 
+        if ($classReflection->hasTraitUse($objectType->getClassName())) {
```

Comment:
just guessing, this probably needs to be on the last check, that cover fallback only, eg:

```php
if (classReflection->is($objectType->getClassName())) {
    return true;
}

return $classReflection->hasTraitUse($objectType->getClassName());
```

also, please run:

```bash
bin/rector
```

to ensure no regression in rector-src codebase, thank you.

Reply:
Thanks for the quick review. I have adjusted the code according to your suggestion.
Also `bin/rector` returns no errors or problems.